web-module/src/main/webapp/resources/structEditJS/autocomplete.mjs

```diff
@@ -139,35 +139,95 @@ class CelAutocompleteInitialiser {
     selectElem.innerHTML = '<option selected="selected" value="">delete</option>';
   }
   
+  #handleAddNewMessage(event, selectElem) {
+    console.debug('addNewButton popup returned', event.data);
+    event.source?.close();
+    const option = new Option(event.data.text, event.data.id, true, true);
+    selectElem.append(option);
```

Comment:
it needs to replace preexisting options, otherwise multiple values will be transmitted.
```suggestion
    selectElem.replaceChildren(new Option(event.data.text, event.data.id, true, true));
```